app/views/hud_reports/_view_filter.haml

```diff
@@ -0,0 +1,51 @@
+:ruby
+  filter_class = if @active_filter then 'filter-active' else 'filter' end
+  report_version = link_params.try(:[], :filter).try(:[], :report_version)
+  if report_version
+    path_for_clear_view_filter = "#{path_for_clear_view_filter}?filter[report_version]=#{report_version}"
```

Comment:
This is bound to end in disaster.  `path_for_clear_view_filter` should take an argument, then just pass in `{filter: {report_version: report_version}}` and have it pass that along to the method it calls.  Manually building the query string won't work if `path_for_clear_view_filter` ever returns a url with existing params.